File: backend/api/schemas/payment.py
"""Pydantic schemas for payment endpoints."""

from __future__ import annotations

from datetime import date

from pydantic import BaseModel


class PaymentResponse(BaseModel):
    id: str
    member_id: str
    member_name: str
    amount: float
    due_date: date
    paid_at: date | None
    status: str


class PaymentListResponse(BaseModel):
    payments: list[PaymentResponse]
    total: int
    page: int
    page_size: int

File: backend_v2/src/domain/entities/payment.ts
import { PaymentStatus } from '../enums';

export interface Payment {
  id: string;
  memberId: string;
  gymId: string;
  dueDate: Date;
  paidAt: Date | null;
  amount: number;
  status: PaymentStatus;
  createdAt: Date;
}
